### DISCUSSION TITLE
Grid JSON Field

### DISCUSSION THREAD

**Discussion by lessborders** [directus/directus#15630] [Feature Requests]
At the moment for my frontend I am using the repeater field to dynamically mount my components. With this I am able to create some pretty cool page layouts, with nested repeaters for nesting components.

The only downside is that its 1. very ugly in the backend and 2. not very flexible when it comes to complex layouts.

So my idea is to be able to have some kind of Grid repeater in the backend, so instead of only being able to create rows, we can also create columns.
Or something like this. Maybe even a "Builder"  field that let's you dynamically add fields like Repeaters, Images, Inputs to a JSON field.

**Comment by benhaynes** [comment]
Hey! Sorry, should have commented over here... a more appropriate discussion for this topic. Very interested to figure out a good solution for this block interface!!

**Reply by benhaynes** [reply]
Ref: https://github.com/directus/directus/discussions/15629

**Reply by lessborders** [reply]
As a quick workaround, I've set up a repeater field that nests 3 levels like so:
![image](https://user-images.githubusercontent.com/100006598/190884178-b4cb343d-16fa-4040-828b-e3a280ce8d25.png)

With this I can take advantage of [grid-template-columns](https://developer.mozilla.org/en-US/docs/Web/CSS/grid-template-columns) to achieve a grid system.

**Reply by lessborders** [reply]
So I would say the necessary fields for a layout editor would be:

1 - Component [string or relationship?]
2 - Props (class, style, custom defined props)  - [JSON or repeater]
3 - Either a "row layout" on row, or a "column width" on the columns, both are flexible. [undecided]
4 - Children/Content [M2M/M2A, repeater or something new?]

**Reply by lessborders** [reply]
If we have a components collection, or whole section for our components, then we would only need Rows, Columns and M2M components on each column

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
